src/utils/contribution.js

```diff
@@ -60,51 +85,228 @@ export default class Contribution {
    * @param  {} changes object containing changes
    * @param  {} field field to check
    * @param  {} value value to apply to the field
+   * @param  {} type used to specify a type of field (e.g. array) to perform different kind of operations
    */
-  static onChange(component, changes, field, value) {
-    const isChanged = !isEqual(value, this.getOriginalValue(component, field))
+  static applyChanges(component, changes, field, value, type, transform = a => a) {
+    const proposedValue = transform(value)
+    const isChanged = !isEqual(proposedValue, this.getOriginalValue(component, field))
     const newChanges = { ...changes }
-    if (isChanged) newChanges[field] = value
+    if (isChanged)
+      type === 'array'
+        ? (newChanges[field] = proposedValue.replace(/\s/g, '').split(','))
+        : (newChanges[field] = proposedValue)
     else delete newChanges[field]
     return newChanges
   }
 
   static getOriginalValue(component, field) {
-    return get(component, field)
+    return get(component, field) || ''
+  }
+
+  static getUpdatedValue(preview, field) {
+    return get(preview, field) || ''
   }
 
   /**
    * Get the value of the specific field into the definition
    * Returns the updated value or the original one if not modified
-   * @param  {} component original component
-   * @param  {} changes object containing changes
-   * @param  {} field field to check
+   * @param  {Object} component original component
+   * @param  {Object} preview updated component sent back from the API after each change
+   * @param  {String} field field to check
+   */
+  static getValue(component, preview, field) {
+    return preview && this.getUpdatedValue(preview, field)
+      ? this.getUpdatedValue(preview, field)
+      : this.getOriginalValue(component, field) || ''
+  }
+
+  /**
+   * Get the value AND if it was Updated of the specific field into the definition
+   * Returns the updated value or the original one if not modified
+   *
+   * @param  {Object} component original component
+   * @param  {Object} preview updated component sent back from the API after each change
+   * @param  {String} field field to check
    */
-  static getValue(component, changes, field) {
-    return changes && changes[field] ? changes[field] : this.getOriginalValue(component, field) || ''
+  static getValueAndIfDifferent(component, preview, field) {
+    const originalValues = get(component, field)
```

Comment:
consider

```javascript
static getValueAndIfDifferent(component, preview, field) {
    if (preview && get(preview, field)) return { value: get(preview, field), isDifferent: true }
    const originalValues = get(component, field)
    if (!originalValues || !originalValues.length) return [{ value: 'core', isDifferent: false }]
    return originalValues.map(value => ({ value: value || 'core', isDifferent: false }))
  }
```

Note that superficially not all cases return arrays. Not sure what is intendedn

Reply:
Thank you, actually i needed to insert some improvements to this function, so i've take the chance to rewrite it in a more proper way